### PR TITLE
Blue Button

### PR DESCRIPTION
**Pull request by Moejoe90**
Started implementing a react blue button component that will need to be used with containers.

### CODE REVIEW

File: index CPY.html [hunk 1]
```diff
@@ -0,0 +1,133 @@
+<!doctype html>
```
Comment (banciur) [inline index CPY.html]: I don't think we should put into repository that kind of files used in development.

File: index ORG.html [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<!doctype html>
```
Comment (banciur) [inline index ORG.html]: I don't think we should put into repository that kind of files used in development.

File: package.json [hunk 3]
```diff
@@ -47,6 +47,7 @@
   "dependencies": {
     "bn": "^1.0.1",
     "ethereumjs-util": "^5.1.1",
+    "flexboxgrid": "^6.3.1",
```
Comment (banciur) [inline package.json]: You added flexboxgrid in two ways one as npm dependency and second as direct stylesheet link in `./public/index.html`
We also already have dependency from `react-flexbox-grid` in `package.json` which is [react implementation](https://github.com/roylee0704/react-flexbox-grid) of flexbox-grid
In my opinion we should choose one way to do it. Or for now if its easier to use classNames use just one dependency from flexboxgrid or direct import in index.html

File: public/index.html [hunk 4]
```diff
@@ -15,8 +15,10 @@
     -->
     <title>To do</title>
     <link href="https://fonts.googleapis.com/css?family=Roboto" rel="stylesheet">
+    <link href="https://fonts.googleapis.com/css?family=Montserrat:400,700" rel="stylesheet">
     <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
     <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
+    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/flexboxgrid/6.3.1/flexboxgrid.css" type="text/css">
```
Comment (banciur) [inline public/index.html]: Take a look at my comment to 'package.json' file

File: public/index.html [hunk 5]
```diff
@@ -15,8 +15,10 @@
     -->
     <title>To do</title>
     <link href="https://fonts.googleapis.com/css?family=Roboto" rel="stylesheet">
+    <link href="https://fonts.googleapis.com/css?family=Montserrat:400,700" rel="stylesheet">
```
Comment (banciur) [inline public/index.html]: There is a more efficient way to import multiple fonts from 'font.googleapis.com'. You could ask for more then one font in single link separating fonts by `|` symbol ex:
`https://fonts.googleapis.com/css?family=Roboto|Montserrat:400,700`
Browser will make one request instead of two.

File: src/components/App.css [hunk 6]
```diff
@@ -18,6 +18,12 @@
   font-size: large;
 }
 
+.rectangle-copy {
```
Comment (banciur) [inline src/components/App.css]: Do we need this css class here? I don't see any element of with class `rectangle-copy `

File: src/components/App.css [hunk 7]
```diff
@@ -18,6 +18,12 @@
...
+  height: 50px;
+  width: 200px;
+  border-radius: 4px;
+  background-c}
```
Comment (banciur) [inline src/components/App.css]: It is error

File: src/components/App.js [hunk 8]
```diff
@@ -14,10 +14,13 @@ import Web3ChainInfo from './Web3ChainInfo';
 import Web3WalletInfo from './Web3WalletInfo';
 import Transactions from './Transactions';
 import TransferForm from './TransferForm';
+import buttonME from './buttonME';
 
 export const AppComponent = ({ i18n }) => (
   <div>
     <AppBar title={i18n('To do')} iconClassNameRight="muidocs-icon-navigation-expand-more" />
+    <buttonME />
+    <submit type="submit">Click Me!</submit>
```
Comment (banciur) [inline src/components/App.js]: `<submit type="submit">Click Me!</submit>` I think we don't need it at this test

File: src/components/bluebutton.js [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+  textalign: 'center',
+  height: '40px',
+  lineheight: '40px',
+};
+
+const Button = React.createClass({
+  render() {
+    return (
+      <button
+        className="btn btn-default"
```
Comment (banciur) [inline src/components/bluebutton.js]: Here you restyled button class that is taken from http://getbootstrap.com/ framework. Plan it to remove it from project.

**Review by banciur — CHANGES_REQUESTED** [review]
I have two main concerns:
- You restyled button from bootstrap. We will remove it from project.
- You committed few development files that should not land in repository

File: package.json (hunk 3 above)
Comment (Moejoe90) [reply package.json]: Your right, i think maybe the best option would be using react as a dependency but what do you think?

File: src/components/App.css (hunk 7 above)
Comment (Moejoe90) [reply src/components/App.css]: Done

File: src/components/App.css (hunk 6 above)
Comment (Moejoe90) [reply src/components/App.css]: Done

File: public/index.html (hunk 4 above)
Comment (Moejoe90) [reply public/index.html]: Removed this and will use react-flex instead

File: public/index.html (hunk 5 above)
Comment (Moejoe90) [reply public/index.html]: Done

File: index CPY.html (hunk 1 above)
Comment (Moejoe90) [reply index CPY.html]: done

File: index ORG.html (hunk 2 above)
Comment (Moejoe90) [reply index ORG.html]: done

File: src/components/App.js (hunk 8 above)
Comment (Moejoe90) [reply src/components/App.js]: done

File: package.json (hunk 3 above)
Comment (banciur) [reply package.json]: Ahh yes that was third thing I wanted to show you in my other PR. We can use react version if there problems with it we can always go back to css.  
Here is [example](https://github.com/Neufund/generic-ico/blob/basic_layouts/src/components/HeaderInitial.js#L11)

File: .travis.yml [hunk 10]
```diff
@@ -5,10 +5,14 @@ notifications:
   email: false
 node_js:
   - "7"
+notifications:
+  email: false
```
Comment (recmo) [inline .travis.yml]: Why is this added?

File: package.json (hunk 3 above)
Comment (recmo) [reply package.json]: I like the react version, makes the code look a bit cleaner.

File: src/components/App.css [hunk 11]
```diff
@@ -0,0 +1,24 @@
+.App {
```
Comment (recmo) [inline src/components/App.css]: @banciur Deleted this file, here it is accidentally created again.

File: src/components/bluebutton.js [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+  margin: '10px 10px 10px 0',
+  width: '100%',
+  backgroundColor: '#09628D',
+  color: 'white',
+  textalign: 'center',
+  height: '40px',
+  lineheight: '40px',
+};
+
+const Button = React.createClass({
```
Comment (recmo) [inline src/components/bluebutton.js]: The filename should be the same as the component name.

File: src/components/buttonME.js [hunk 13]
```diff
@@ -0,0 +1,5 @@
+import React from 'react';
+
+const buttonME = () => <span> test</span>;
```
Comment (recmo) [inline src/components/buttonME.js]: Component names must start with a Capital letter.  (If you don't, React will think you are using an html tag. For example, `<buttonME/>` will not render correctly.)

File: src/components/htmllayout.js [hunk 14]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
```
Comment (recmo) [inline src/components/htmllayout.js]: Is this a file for testing? If so, it should go. Layout is being done by @banciur.

File: src/routes.js [hunk 15]
```diff
@@ -60,6 +60,7 @@ const routes = {
   '/register/email/:confirmationId': ConfirmationEmailPage,
   '/kyc': KYCPage,
   '/not-found': NotFoundPage,
+
```
Comment (recmo) [inline src/routes.js]: There should not be a newline here.

File: .travis.yml [hunk 16]
```diff
@@ -5,10 +5,14 @@ notifications:
...
 cache: yarn
 install:
   - yarn
 script:
   - yarn test
   - yarn run -- codecov
   - yarn lint
+
+
```
Comment (recmo) [inline .travis.yml]: There should be no additional newlines here.

File: src/components/bluebutton.js [hunk 17]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+
+const buttonStyle = {
```
Comment (recmo) [inline src/components/bluebutton.js]: Styles should go in their own `.scss` file. Look at `Footer` and `HeaderInitial` for examples.

File: src/components/bluebutton.js (hunk 9 above)
Comment (recmo) [reply src/components/bluebutton.js]: Agreed. We are dropping bootstrap. In fact, I think we didn't have it all? Where's this class coming from?

Update: Yes. Bootstrap is gone, these classes don't do anything anymore and should be removed.

**Review by recmo — CHANGES_REQUESTED** [review]

File: src/components/App.css (hunk 11 above)
Comment (Moejoe90) [reply src/components/App.css]: Done, i think there might have a bin a mistake rebase

File: src/components/bluebutton.js [hunk 18]
```diff
@@ -0,0 +1,36 @@
...
+    className={styles.our_button}
+    role="button"
+    tabIndex="-1"
+    {...props}
+    onClick={(event) => {
+      event.preventDefault();
+      targetTo();
+    }}
+  >
+    {label}
```
Comment (recmo) [inline src/components/bluebutton.js]: I'ts better to use [`children`](https://facebook.github.io/react/docs/composition-vs-inheritance.html#containment) for this, than you can use the button as `<Button>Text</Button>`.

File: src/components/bluebutton.js [hunk 19]
```diff
@@ -0,0 +1,36 @@
...
+
+const Button = ({ targetTo, label, ...props }) =>
+(
+  <div
+    className={styles.our_button}
+    role="button"
+    tabIndex="-1"
+    {...props}
+    onClick={(event) => {
+      event.preventDefault();
```
Comment (recmo) [inline src/components/bluebutton.js]: I don't think we need `preventDefault` here. We can do something like `onClick={()=>onClick()}` or even `onClick={onClick}`.

@banciur Do you know what's best practice here?

File: src/components/bluebutton.js [hunk 20]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import styles from './bluebutton.scss';
+
+const Button = ({ targetTo, label, ...props }) =>
+(
+  <div
```
Comment (recmo) [inline src/components/bluebutton.js]: `<div>` is really generic and should not be used if a more descriptive element exists. In this case [`<button>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button) is a better start.

File: src/components/bluebutton.js [hunk 21]
```diff
@@ -0,0 +1,36 @@
...
+  </div>
+  );
+
+Button.propTypes = {
+  targetTo: PropTypes.func.isRequired,
+  label: PropTypes.string.isRequired,
+};
+
+Button.defaultProps = {
+  label: 'Name me!',
```
Comment (recmo) [inline src/components/bluebutton.js]: This is not a useful default, you also don't need a default because you made `label` a required property.

File: src/components/bluebutton.js [hunk 22]
```diff
@@ -0,0 +1,36 @@
...
+};
+
+module.exports = Button;
+
+// todo: Add Flex box integration
+//       Add SASS implemenation
+//       Add additional abstractions
+//       Look into containers or stand alone
```
Comment (recmo) [inline src/components/bluebutton.js]: I think some of these Todo's are outdated.
They should also all be of the format `// TODO: <text>` so that [automated tools](https://www.npmjs.com/package/code-notes) can find them.

File: src/components/bluebutton.scss [hunk 23]
```diff
@@ -0,0 +1,13 @@
+.our_button {
```
Comment (recmo) [inline src/components/bluebutton.scss]: The class name should be the same as the component name.

**Review by recmo — CHANGES_REQUESTED** [review]

File: src/components/bluebutton.js (hunk 20 above)
Comment (Moejoe90) [reply src/components/bluebutton.js]: the button tag a bit different in the way it looks, it will require color change, and overriding some default params

File: src/components/bluebutton.js (hunk 21 above)
Comment (Moejoe90) [reply src/components/bluebutton.js]: That's true

File: src/components/bluebutton.scss (hunk 23 above)
Comment (Moejoe90) [reply src/components/bluebutton.scss]: On that

File: src/components/bluebutton.js (hunk 19 above)
Comment (recmo) [reply src/components/bluebutton.js]: I read [the manual](https://facebook.github.io/react/docs/handling-events.html) and came to the conclusion that `onClick={onClick]` is *wrong* (the `this` will be rebound, leading to weird bugs).

`event.preventDefault()` is not necessary here, because a plain button does not have a default.

So I suggest `onClick={(e)=>onClick(e)}`. Like the last example in the manual. The `onClick` handler can choose to ignore the `e` argument.

Comment (Moejoe90) [reply src/components/bluebutton.js]: Ah that makes sense as its not an anchor with default activity of for example becoming blue after clicking , 
i will implement it like the manuel

File: src/components/bluebutton.js (hunk 18 above)
Comment (Moejoe90) [reply src/components/bluebutton.js]: done

File: src/components/bluebutton.js (hunk 22 above)
Comment (Moejoe90) [reply src/components/bluebutton.js]: done

File: src/components/bluebutton.js [hunk 24]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
```
Comment (recmo) [inline src/components/bluebutton.js]: Filename `bluebutton.js` should be the same as component name `Button`.

File: src/components/bluebutton.js [hunk 25]
```diff
@@ -0,0 +1,27 @@
...
+import PropTypes from 'prop-types';
+
```
Comment (recmo) [inline src/components/bluebutton.js]: There should not be a newline here.

File: src/components/bluebutton.js [hunk 26]
```diff
@@ -0,0 +1,27 @@
...
+import styles from './bluebutton.scss';
+
+const Button = ({ targetTo, children, label, ...props }) =>
+(
+  <button
+    className={styles.Button}
+    tabIndex="-1"
+    {...props}
+    onClick={() => { targetTo(); }}
```
Comment (recmo) [inline src/components/bluebutton.js]: You can remove the `;`.  If you do that, you can also remove the `{` and `}` arround `targetTo()`.

File: src/components/bluebutton.js [hunk 27]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import styles from './bluebutton.scss';
+
+const Button = ({ targetTo, children, label, ...props }) =>
```
Comment (recmo) [inline src/components/bluebutton.js]: `targetTo` is not a descriptive name for the button's on click handler. Why not `onClick` ?

Comment (recmo) [inline src/components/bluebutton.js]: The (required) property `label` is never used, it should be removed.

File: src/components/bluebutton.js [hunk 28]
```diff
@@ -0,0 +1,27 @@
...
+Button.propTypes = {
+  targetTo: PropTypes.func.isRequired,
+  label: PropTypes.string.isRequired,
+  children: PropTypes.node.isRequired,
+};
+
+module.exports = Button;
+
+// TODO: <Add additional abstractions colors modes>
+// TODO: <Look into containers or stand alone>
```
Comment (recmo) [inline src/components/bluebutton.js]: I think I was unclear. The format is `// TODO: Your text goes here`. The `<` and `>` are not required.

**Review by recmo — CHANGES_REQUESTED** [review]

File: src/components/bluebutton.js (hunk 27 above)
Comment (Moejoe90) [reply src/components/bluebutton.js]: From my point its the function that gets called when the button event happens,

Comment (Moejoe90) [reply src/components/bluebutton.js]: done

File: src/components/bluebutton.js (hunk 26 above)
Comment (Moejoe90) [reply src/components/bluebutton.js]: done

File: src/components/bluebutton.js (hunk 28 above)
Comment (Moejoe90) [reply src/components/bluebutton.js]: ah i get your point

File: src/components/bluebutton.js (hunk 24 above)
Comment (Moejoe90) [reply src/components/bluebutton.js]: done

File: src/components/bluebutton.js (hunk 25 above)
Comment (Moejoe90) [reply src/components/bluebutton.js]: done

File: src/components/bluebutton.js (hunk 27 above)
Comment (recmo) [reply src/components/bluebutton.js]: Exactly. Why would you call this fuction `targetTo`?  I know it's copied from `Link`, and for link it sort-of made sense to call it `targetTo`. Here it doesn't make much sense.

This is a name you can choose, and you can choose it to be anything you want. You can keep the name `targetTo` but you can also change it into something else.

**Review by recmo — APPROVED** [review]

**Review by banciur — APPROVED** [review]
